docs/registry/attributes/http.md

```diff
@@ -79,6 +79,17 @@ Examples:
 **[5] `http.route`:** MUST NOT be populated when this is not supported by the HTTP server framework as the route attribute should have low-cardinality and the URI path can NOT substitute it.
 SHOULD include the [application root](/docs/http/http-spans.md#http-server-definitions) if there is one.
 
+A static path segment is a part of the route template with a fixed, low-cardinality value. This includes literal strings like `/users/` and placeholders that are constrained to a finite, predefined set of values `{controller}` or `{action}`.
+
+A dynamic path segment is a placeholder for a value that can have high cardinality and is not constrained to a predefined list like `{id}`.
+
+Examples of converting route templates to `http.route` values:
+
+* ASP.NET: `{controller}/{action}/{id?}` becomes `/users/get/{id}`
+* Django: `/articles/int:id/edit/` becomes `/articles/:id/edit`
```

Comment:
what would be the benefit of this conversion for the end users? who would do it?

keep in mind, this is stable spec and implemented by stable instrumentation. We don't guarantee that attribute values remain stable, but there should still be a strong reason to change something that's stable.

Reply:
`The span name SHOULD be the most general string that identifies a (statistically) interesting class of Spans`

:int as a constraint is not the most general string. Although one can argue that if you have routes:
/articles/int:id/edit/
/articles/string:id/edit/

... then those should be treated as separate '(statistically) interesting class of Spans'. Which then breaks the rule of course that be the most general string. Constraints can add some context.

Reply:
So you'd like otel instrumentations to have a freedom to format routes as they see fit to optimize user experience while staying within the spec boundaries? That's fine, we can put this language in the spec.  I don't believe nothing in HTTP semconv contradicts it though.

Specifying how exactly it should be done would be hard and would need extensive research across web frameworks and design. I don't see a huge benefit given that it very fx and instrumentation-specific.